### PR TITLE
[WIP] Add LVM driver support

### PR DESCRIPTION
**Pull request by leonwanghui**
In this patch, I added the partial implementation of lvm driver (volume CRUD, snapshot CRUD). And to compatible with current framework, I did a bit change of OpenSDS model, which is adding ```Metadata``` field in VolumeSpec and VolumeSnapshotSpec.

### CODE REVIEW

**Comment by leonwanghui** [comment]
Add a reference with Issue [#116](https://github.com/opensds/opensds/issues/116)

File: contrib/drivers/lvm/lvm.go [hunk 1]
```diff
@@ -0,0 +1,290 @@
...
+}
+
+type LVMConfig struct {
+	Pool map[string]PoolProperties `yaml:"pool,flow"`
+}
+
+type PoolProperties struct {
+	DiskType  string `yaml:"diskType"`
+	IOPS      int64  `yaml:"iops"`
+	BandWidth int64  `yaml:"bandwidth"`
```
Comment (wisererik) [inline contrib/drivers/lvm/lvm.go]: BandWidth or Bandwidth?

**Comment by wisererik** [comment]
Hi, Leon, can the binary files in the integration test directory be automatically generated when the test is excuted?

File: contrib/drivers/lvm/lvm.go (hunk 1 above)
Comment (leonwanghui) [reply contrib/drivers/lvm/lvm.go]: It should be Bandwidth, but I didn't have so much energy to solve it because it's a bit complicated.

**Comment by leonwanghui** [comment]
@wisererik Yeah, you can modify it in travis.yml file.

**Pull request merged**